pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -0,0 +1,151 @@
+package trustedcabundle
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsci "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+)
+
+const (
+	injectionOfCABundleAnnotatoion = "security.opendatahub.io/inject-trusted-ca-bundle"
+	CAConfigMapName                = "odh-trusted-ca-bundle"
+	CADataFieldName                = "odh-ca-bundle.crt"
+)
+
+func ShouldInjectTrustedBundle(ns client.Object) bool {
+	if !strings.HasPrefix(ns.GetName(), "openshift-") && !strings.HasPrefix(ns.GetName(), "kube-") &&
+		ns.GetName() != "default" && ns.GetName() != "openshift" && !HasCABundleAnnotationDisabled(ns) {
+		return true
+	}
+	return false
+}
+
+func HasCABundleAnnotationDisabled(ns client.Object) bool {
+	if value, found := ns.GetAnnotations()[injectionOfCABundleAnnotatoion]; found {
+		enabled, err := strconv.ParseBool(value)
+		return err == nil && !enabled
+	}
+	return false
+}
+
+func AddCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsci.DSCInitialization) error {
+	namespaceList := &corev1.NamespaceList{}
+	err := cli.List(ctx, namespaceList)
+	if err != nil {
+		return err
+	}
+
+	for i := range namespaceList.Items {
+		ns := &namespaceList.Items[i]
+		if ShouldInjectTrustedBundle(ns) {
+			if err := CreateOdhTrustedCABundleConfigMap(ctx, cli, ns.Name, dscInit); err != nil {
+				return err
```

Comment:
question: should we return if the operation fails for a single namespace? or should we try to add to all namespaces and collect a list of errors returning at the end? I am not sure about it.